test/e2e/workload_test.go

```diff
@@ -194,8 +201,14 @@ var _ = Describe("Deployment", func() {
 			err = framework.WaitUntilDeploymentDeleted(f.KubeClient, deployment.ObjectMeta)
 			Expect(err).NotTo(HaveOccurred())
 
+			f.DeleteRepositories(f.DeploymentRepos(&deployment))
+			err = framework.WaitUntilRepositoriesDeleted(f.StashClient, f.DeploymentRepos(&deployment))
```

Comment:
Why `f.DeploymentRepos(&deployment)` ? Now, we create `Repository` our self so we know the repository name. There is no necessary to detecting it using another function.